src/api/java/genkinds.py.in

```diff
@@ -77,15 +79,47 @@ KINDS_JAVA_BOTTOM = \
 }
 """
 
+# mapping from c++ patterns to java
+CPP_JAVA_MAPPING = \
+    {
+        r"\bbool\b": "boolean",
+        r"\bconst\b\s?": "",
+        r"\bint32_t\b": "int",
+        r"\bint64_t\b": "long",
+        r"\buint32_t\b": "int",
+        r"\buint64_t\b": "long",
+        r"\bunsigned char\b": "byte",
+        r"cvc5::api::": "cvc5.",
+        r"Term::Term\(\)": "Solver.getNullTerm()",
+        r"Solver::": "Solver.",
+        r"std::vector<int>": "int[]",
+        r"std::vector<Term>": "Term[]",
+        r"std::string": "String",
+        r"&": "",
+        r"::": "."
+    }
 
-# Files generation
 
+# convert from c++ doc to java doc
+def format_comment(comment):
+    for kind, value in CPP_JAVA_MAPPING.items():
```

Comment:
Minor nitpick, but kind makes me thing that this is a CVC4 kind, but it's actual a string representing a C++ type, right? Perhaps `cpptype` or something like that would be a better name. And then `value` could be `javatype` or something.

Reply:
Copy & paste problem. Thanks for pointing this out. I guess pattern and replacement are better here because key is not always a `cpptype`.